plugin-vidarr/src/main/java/ca/on/oicr/gsi/shesmu/vidarr/VidarrPlugin.java

```diff
@@ -50,6 +54,95 @@
 
 public class VidarrPlugin extends JsonPluginFile<Configuration> {
 
+  private class AnalysisCache extends ValueCache<Stream<VidarrAnalysisValue>> {
+    public AnalysisCache(Path fileName) {
+      super("vidarr-analysis " + fileName.toString(), 30, ReplacingRecord::new);
+    }
+
+    private String createVidarrProvenanceRequestBody(List<String> versionTypes) {
+      return "{"
+          + "\"analysisTypes\": [\"FILE\"],"
+          + "\"epoch\": 0,"
+          + "\"includeParameters\": false,"
+          + "\"timestamp\": 0,"
+          + "\"versionPolicy\": \"LATEST\","
+          + "\"versionTypes\": ["
+          + versionTypes.stream().map(name -> ("\"" + name + "\"")).collect(Collectors.joining(","))
+          + "  ]}";
+    }
+
+    protected Stream<VidarrAnalysisValue> analysisArchive(String baseUrl) throws Exception {
+
+      if (configuration.isEmpty()) {
+        return new ErrorableStream<>(Stream.empty(), false);
+      }
+
+      final var results =
+          HTTP_CLIENT.send(
+              HttpRequest.newBuilder(URI.create(baseUrl + "/api/provenance"))
+                  .header("Content-type", "application/json")
+                  .timeout(Duration.ofMinutes(10))
+                  .POST(
+                      HttpRequest.BodyPublishers.ofString(
+                          createVidarrProvenanceRequestBody(configuration.get().getVersionTypes())))
+                  .build(),
+              new JsonBodyHandler<>(
+                  MAPPER, new TypeReference<AnalysisProvenanceResponse<ExternalKey>>() {}));
```

Comment:
The [VersionPolicy documentation](https://github.com/oicr-gsi/vidarr/blob/master/vidarr-pluginapi/src/main/java/ca/on/oicr/gsi/vidarr/api/VersionPolicy.java#L14) says that the type argument for AnalysisProvenanceResponse should differ depending on the VersionPolicy. Unfortunately, the different types (ExternalId, ExternalKey, MultiVersionExternalKey) would need to have different ShesmuVariable types as they have different version representations, which gets messy really quickly. Given that we're expecting to always use the LATEST version policy, what do y'all think about hard-coding that (so, removing it from configuration) and keeping this type argument as ExternalKey?

Reply:
The version policy is already hardcoded: https://github.com/oicr-gsi/shesmu/pull/1350/files#diff-5e0eae5036eb27102f91d90877b5132685e42e8990c087c3bee9c73b7330dd73R68